docs/markdown/Writing Plugins/tutorials/01-create-new-goal-with-a-custom-target.md

```diff
@@ -0,0 +1,335 @@
+# Introduction
+
+In this tutorial, you'll learn the basics needed to get started writing a plugin. You will create a new goal, `project-version`, which will tell you the version (retrieved from the `VERSION` text file) of a particular project in your monorepository. You will learn how to create a new custom target to refer to the `VERSION` file, how to author a new goal, and, most importantly, how to connect rules and targets. You can follow along this tutorial in your own repository; you only need to be on a recent version of Pants and have a `VERSION` file containing a version string e.g. `1.2.3`.
+
+## Registering a plugin
+
+We'll be writing an [in-repo plugin](https://www.pantsbuild.org/docs/plugins-overview#in-repo-plugins), and expect you to have the `pants-plugins/project_version` directory with the `pants.toml` configuration file extended:
+
+```toml pants.toml
+pythonpath = ["%(buildroot)s/pants-plugins"]
+
+backend_packages = [
+  "pants.backend.python",
```

Comment:
I think we still need `"pants.backend.plugin_development"` as well?

Reply:
According to the docs, it's optional. I've been able to develop a plugin in a local repository without this backend enabled, so perhaps we could leave it out to keep the code minimal for the very first tutorial and introduce it in the upcoming one? I plan to write more as the next tutorial will build upon the outcome of the first one.

Reply:
yea, currently it's only for the `pants_requirements` target, so if you don't use that.. :)

Reply:
okay, I'll leave it out in this particular, entry level tutorial.